### PR TITLE
Support group_instance_id on skip conditions for groups

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Skip conditions are now supported in repeating groups to conditionally not show one of the repeats.

### How to review 
Launch schema `test_routing_on_answer_from_driving_repeating_group.json` and observe that when answering that you do not live here then you are not included in the repeating group being asked their gender.

### CODE REVIEW

**Comment by dcdarrell9** [comment]
screen shot 2018-08-30 at 10 40 01
I answered for 5 people and looks like it's cutting off past 2 and not displaying name correctly in the `What is 's sex?` question

File: app/questionnaire/group_dependencies.py [hunk 1]
```diff
@@ -64,15 +64,41 @@ def get_group_dependencies(schema):
...
-                            _get_dependency_drivers_from_group_ids(group['id'],
-                                                                   routing_rule['repeat']['group_ids']))
-                    else:
-                        dependencies.update(
-                            _get_dependency_driver_from_answer_id(schema,
-                                                                  group['id'],
-                                                                  routing_rule['repeat']['answer_id']))
+                    dependencies.update(_build_routing_dependencies(schema, routing_rule, group['id']))
+
+                    # Only when the group is a repeat
```
Comment (insacuri) [inline app/questionnaire/group_dependencies.py]: I don't think the comment is helpful as there isn't a conditional around the next line.  Either expand this comment to give more information or remove it.

File: app/questionnaire/rules.py [hunk 2]
```diff
@@ -116,26 +116,26 @@ def evaluate_repeat(schema, repeat_rule, answer_store, answer_ids_on_path):
...
-        no_of_repeats = len(group_instances)
     else:
         repeat_functions = {
             'answer_value': _get_answer_value,
             'answer_count': len,
             'answer_count_minus_one': _get_answer_count_minus_one,
         }
 
-        repeat_index = repeat_rule['answer_id']
+        repeat_indexex = []
```
Comment (insacuri) [inline app/questionnaire/rules.py]: Is this name intentional?

**Review by insacuri — CHANGES_REQUESTED** [review]
Other than my 2 fairly small comments, looks good to me

**Comment by jonnyshaw89** [comment]
@dcdarrell9 that is another piece of work that @dwyeradam is working on.

File: app/questionnaire/rules.py (hunk 2 above)
Comment (jonnyshaw89) [reply app/questionnaire/rules.py]: I just cant spell

**Review by insacuri — APPROVED** [review]

File: app/questionnaire/rules.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def convert_to_datetime(value):
     return datetime.strptime(value, date_format) if value else None
 
 
-def evaluate_goto(goto_rule, schema, metadata, answer_store, group_instance):
+def evaluate_goto(goto_rule, schema, metadata, answer_store, group_instance, group_instance_id=None):
```
Comment (bitdivision) [inline app/questionnaire/rules.py:95]: Pedantic: Add group_instance_id to docstring.

File: app/questionnaire/rules.py [hunk 4]
```diff
@@ -155,7 +155,7 @@ def _get_answer_count_minus_one(filtered_answers):
     return min(max(len(filtered_answers) - 1, 0), MAX_REPEATS - 1)
 
 
-def evaluate_skip_conditions(skip_conditions, schema, metadata, answer_store, group_instance=0):
+def evaluate_skip_conditions(skip_conditions, schema, metadata, answer_store, group_instance=0, group_instance_id=None):
```
Comment (bitdivision) [inline app/questionnaire/rules.py:159]: Pedantic: docstring.

File: app/questionnaire/rules.py [hunk 5]
```diff
@@ -171,13 +171,13 @@ def evaluate_skip_conditions(skip_conditions, schema, metadata, answer_store, gr
...
     for when in skip_conditions:
-        condition = evaluate_when_rules(when['when'], schema, metadata, answer_store, group_instance)
+        condition = evaluate_when_rules(when['when'], schema, metadata, answer_store, group_instance, group_instance_id)
         if condition is True:
             return True
     return False
 
 
-def evaluate_when_rules(when_rules, schema, metadata, answer_store, group_instance):
+def evaluate_when_rules(when_rules, schema, metadata, answer_store, group_instance, group_instance_id):
```
Comment (bitdivision) [inline app/questionnaire/rules.py:182]: Pedantic: docstring.

**Review by bitdivision — APPROVED** [review]
Minor pedantic docstring comments. Code looks good.

Will need to remove validator branch before merging.

**Comment by jonnyshaw89** [comment]
@bitdivision cant remove validator branch until https://github.com/ONSdigital/eq-schema-validator/pull/80 is merged

**Pull request merged**